File: 0027-remove-element/0027-remove-element.py
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        k = 0

        for n in nums:
            if n != val:
                k+=1
        l = 0
        for r in range(len(nums)):
            if nums[r] != val:
                nums[l],nums[r] = nums[r],nums[l]
                l += 1
            
                
    
        return k

        

File: 0027-remove-element/0027-remove-element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        vector<int> notValElements;
        int i = 0;

        while (i < nums.size()) {
            if (nums[i] != val) {
                notValElements.push_back(nums[i]);
            }
            i = i + 1;
        }

        int j = 0;
        while (j < notValElements.size()) {
            nums[j] = notValElements[j];
            j = j + 1;
        }

        int originalSize = nums.size();
        int newSize = notValElements.size();
        int difference = originalSize - newSize;

        int counter = 0;
        while (counter < difference) {
            nums[newSize + counter] = val;
            counter = counter + 1;
        }

        return newSize;
    }
};
